src/sentry/dynamic_sampling/rules/utils.py

```diff
@@ -115,6 +142,16 @@ def get_rule_hash(rule: BaseRule) -> int:
     ).__hash__()
 
 
+def get_sampling_value(rule: PolymorphicRule) -> Optional[Tuple[str, float]]:
+    # Gets the sampling value from the rule, based on the type.
+    if "samplingValue" in rule and "value" in rule["samplingValue"]:  # type:ignore
+        return "factor", float(rule["samplingValue"]["value"])  # type:ignore
+    elif "sampleRate" in rule:
+        return "sample_rate", rule["sampleRate"]  # type:ignore
+
+    return None
```

Comment:
Could this be an abstract method on `BaseRule`? If you then implement it on each subtype, the `# type:ignore` should not be necessary.

Reply:
Yep, you could do something like that.